test/test_custom_ops.py

```diff
@@ -678,6 +680,134 @@ def foo(x: Tensor, y: int) -> Tensor:
 
             infer_schema(foo, mutated_args={"y"})
 
+    def test_blackbox_basic(self):
+        @def_blackbox(mutated_args=())
+        def f(x: Tensor, y: float) -> Tensor:
+            return x + y
+
+        x = torch.randn(3)
+        y = 3.14
+        z = f(x, y)
+        self.assertEqual(z, x + y)
+
+        @f.impl("cpu")
+        def _(x, y):
+            return x - y
+
+        z = f(x, y)
+        self.assertEqual(z, x - y)
+
+        with self.assertRaises(RuntimeError):
+            with torch._subclasses.fake_tensor.FakeTensorMode():
+                x = torch.randn(3)
+                f(x, y)
+
+        @f.impl_abstract
+        def _(x, y):
+            return torch.empty_like(x)
+
+        with torch._subclasses.fake_tensor.FakeTensorMode():
+            x = torch.randn(3)
+            z = f(x, y)
+            self.assertEqual(z.shape, x.shape)
+
+    @skipIfTorchDynamo("recursive dynamo")
+    @unittest.skipIf(IS_WINDOWS, "torch.compile doesn't work on windows")
+    @unittest.skipIf(
+        sys.version_info >= (3, 12), "torch.compile is not supported on python 3.12+"
+    )
+    def test_blackbox_compile(self):
+        @torch.library.def_blackbox(mutated_args=())
+        def custom_linear(x: Tensor, weight: Tensor, bias: Tensor) -> Tensor:
+            return (x @ weight.t()) + bias
+
+        @custom_linear.impl_abstract
+        def _(x, weight, bias):
+            assert x.dim() == 2
+            assert weight.dim() == 2
+            assert bias.dim() == 1
+            assert x.shape[1] == weight.shape[1]
+            assert weight.shape[0] == bias.shape[0]
+            assert x.device == weight.device
+            return x.new_empty(x.size(0), weight.size(0))
+
+        x = torch.randn(2, 2)
+        weight = torch.randn(2, 2)
+        bias = torch.randn(2)
+        out = torch.compile(custom_linear, backend="eager", fullgraph=True)(
+            x, weight, bias
+        )
+        assert torch.allclose(out, torch.nn.functional.linear(x, weight, bias))
```

Comment:
Use self.assertEqual? Also, if def_blackbox had done literally nothing, you would still pass this test. You probably need to toggle a flag inside the implementations to test that you've actually exercised the code in question.